### PR TITLE
BIGTOP-3498. Bump Phoenix to 5.1.0.

### PR DESCRIPTION
**Pull request by sekikn**

### CODE REVIEW

**Comment by sekikn** [comment]
I dropped the phoenix-queryserver package and Hive connector jar in this PR, because they have been developed separately from Phoenix core in recent versions of Phoenix.
https://github.com/apache/phoenix-queryserver/
https://github.com/apache/phoenix-connectors/
I'll add them as different components in the future if needed.

----

I confirmed that this PR worked on CentOS 7 and Ubuntu 18.04, as follows:

* After building Phoenix, create a cluster using docker provisioner:

```
$ ./docker-hadoop.sh -d -k 'hdfs, hbase' -L -C config_centos-7.yaml -c 4
```

* Install Phoenix into the cluster manually (this process should be automated by Puppet in the future):

```
$ for i in $(seq 1 4); do
>   ./docker-hadoop.sh --exec $i yum install -y phoenix
>   ./docker-hadoop.sh --exec $i ln -s /usr/lib/phoenix/phoenix-server.jar /usr/lib/hbase/lib/phoenix-server.jar
>   if [ $i = 1 ]; then ./docker-hadoop.sh --exec $i systemctl restart hbase-master; fi
>   ./docker-hadoop.sh --exec $i systemctl restart hbase-regionserver
> done
```

* Make sure that the installed Phoenix works (use `python3 /usr/lib/phoenix/bin/sqlline.py` instead on the distributions that `python` is neither an alias of `python2` nor `python3` by default, such as Ubuntu 18.04):

```
$ ./docker-hadoop.sh --exec 1 /usr/lib/phoenix/bin/sqlline.py

(snip)

Connected to: Phoenix (version 5.1)
Driver: PhoenixEmbeddedDriver (version 5.1)
Autocommit status: true
Transaction isolation: TRANSACTION_READ_COMMITTED
sqlline version 1.9.0
0: jdbc:phoenix:> !table
+-----------+-------------+------------+--------------+---------+-----------+---------------------------+----------------+-------------+----------------+--------------+--------------+----------------+-----------+------------+---------------+------------+
| TABLE_CAT | TABLE_SCHEM | TABLE_NAME |  TABLE_TYPE  | REMARKS | TYPE_NAME | SELF_REFERENCING_COL_NAME | REF_GENERATION | INDEX_STATE | IMMUTABLE_ROWS | SALT_BUCKETS | MULTI_TENANT | VIEW_STATEMENT | VIEW_TYPE | INDEX_TYPE | TRANSACTIONAL | IS_NAMESPA |
+-----------+-------------+------------+--------------+---------+-----------+---------------------------+----------------+-------------+----------------+--------------+--------------+----------------+-----------+------------+---------------+------------+
|           | SYSTEM      | CATALOG    | SYSTEM TABLE |         |           |                           |                |             | false          | null         | false        |                |           |            | false         | false      |
|           | SYSTEM      | CHILD_LINK | SYSTEM TABLE |         |           |                           |                |             | false          | null         | false        |                |           |            | false         | false      |
|           | SYSTEM      | FUNCTION   | SYSTEM TABLE |         |           |                           |                |             | false          | null         | false        |                |           |            | false         | false      |
|           | SYSTEM      | LOG        | SYSTEM TABLE |         |           |                           |                |             | true           | 32           | false        |                |           |            | false         | false      |
|           | SYSTEM      | MUTEX      | SYSTEM TABLE |         |           |                           |                |             | true           | null         | false        |                |           |            | false         | false      |
|           | SYSTEM      | SEQUENCE   | SYSTEM TABLE |         |           |                           |                |             | false          | null         | false        |                |           |            | false         | false      |
|           | SYSTEM      | STATS      | SYSTEM TABLE |         |           |                           |                |             | false          | null         | false        |                |           |            | false         | false      |
|           | SYSTEM      | TASK       | SYSTEM TABLE |         |           |                           |                |             | false          | null         | false        |                |           |            | false         | false      |
+-----------+-------------+------------+--------------+---------+-----------+---------------------------+----------------+-------------+----------------+--------------+--------------+----------------+-----------+------------+---------------+------------+
0: jdbc:phoenix:> 
```

**Review by iwasakims — APPROVED** [review]
+1. This worked on CentOS 8 too.

File: bigtop.bom [hunk 1]
```diff
@@ -304,11 +304,10 @@ bigtop {
        * to the base HBase version. Update as needed whenever changing the
        * HBase version in the BOM.
        */
```
Comment (iwasakims) [inline bigtop.bom:306]: @sekikn You can remove this comment before merging.

Comment (sekikn) [reply bigtop.bom]: Just removed, thanks @iwasakims!

**Pull request merged**